### PR TITLE
switch stcal requirement to dev

### PR DESCRIPTION
**Pull request by braingram**
This PR switches the stcal requirement back to dev. This will allow the changes in https://github.com/spacetelescope/stcal/pull/319 to be picked up by all subsequent romancal runs.

The "dev" regtests from last night all passed. As these used a dev stcal I won't start a separate regtest run for this PR:
https://github.com/spacetelescope/RegressionTests/actions/runs/12000835422/job/33450538484#step:15:124

## Tasks
- [ ] **request a review from someone specific**, to avoid making the maintainers review every PR
- [ ] add a build milestone, i.e. `24Q4_B15` (use the [latest build](https://github.com/spacetelescope/romancal/milestones) if not sure)
- [ ] Does this PR change user-facing code / API? (if not, label with `no-changelog-entry-needed`)
  - [ ] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types)
  - [ ] update or add relevant tests
  - [ ] update relevant docstrings and / or `docs/` page
  - [ ] [start a regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml) and include a link to the running job ([click here for instructions](https://github.com/spacetelescope/RegressionTests/blob/main/docs/running_regression_tests.md))
    - [ ] Do truth files need to be updated ("okified")?
      - [ ] **after the reviewer has approved these changes**, run `okify_regtests` to update the truth files
- [ ] if a JIRA ticket exists, [make sure it is resolved properly](https://github.com/spacetelescope/romancal/wiki/How-to-resolve-JIRA-issues)

<details><summary>news fragment change types...</summary>

  - ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
  - ``changes/<PR#>.docs.rst``
  - ``changes/<PR#>.stpipe.rst``
  - ``changes/<PR#>.associations.rst``
  - ``changes/<PR#>.scripts.rst``
  - ``changes/<PR#>.mosaic_pipeline.rst``
  - ``changes/<PR#>.patch_match.rst``

  ## steps
  - ``changes/<PR#>.dq_init.rst``
  - ``changes/<PR#>.saturation.rst``
  - ``changes/<PR#>.refpix.rst``
  - ``changes/<PR#>.linearity.rst``
  - ``changes/<PR#>.dark_current.rst``
  - ``changes/<PR#>.jump_detection.rst``
  - ``changes/<PR#>.ramp_fitting.rst``
  - ``changes/<PR#>.assign_wcs.rst``
  - ``changes/<PR#>.flatfield.rst``
  - ``changes/<PR#>.photom.rst``
  - ``changes/<PR#>.flux.rst``
  - ``changes/<PR#>.source_detection.rst``
  - ``changes/<PR#>.tweakreg.rst``
  - ``changes/<PR#>.skymatch.rst``
  - ``changes/<PR#>.outlier_detection.rst``
  - ``changes/<PR#>.resample.rst``
  - ``changes/<PR#>.source_catalog.rst``
</details>

### CODE REVIEW

**Review by mairanteodoro — APPROVED** [review]
Thanks, @braingram!

**Comment by braingram** [comment]
Weird. The python 3.12 job used to install setuptools and now does not (leading to a failure due to `pysiaf` importing `pkg_resources` :-/ I guess we could add a setuptools dependency for 3.12?

File: pyproject.toml [hunk 1]
```diff
@@ -26,15 +26,20 @@ dependencies = [
...
     "tweakwcs >=0.8.8",
     "spherical-geometry >= 1.2.22",
     "stsci.imagestats >= 1.6.3",
     "drizzle >= 1.15.0",
     "webbpsf >= 1.2.1",
+    # pysiaf required by webbpsf uses pkg_resources which doesn't
+    # exist in python 3.12 without setuptools. This dependency
+    # can be dropped when webbpsf is dropped (or pysiaf is updated
+    # to not require pkg_resources).
+    "setuptools >= 60 ; python_version >= '3.12'",
```
Comment (zacharyburnett) [inline pyproject.toml:42]: add a TODO flag
```suggestion
    # pysiaf required by webbpsf uses pkg_resources which doesn't
    # exist in python 3.12 without setuptools. 
    # TODO: drop this dependency when webbpsf is dropped 
    # (or when pysiaf no longer requires pkg_resources).
    "setuptools >= 60 ; python_version >= '3.12'",
```

**Review by zacharyburnett — APPROVED** [review]
this LGTM

**Pull request merged**